### PR TITLE
Rewrote tests in Mockery

### LINKED ISSUE #448: Rewrite tests in Mockery
Rewrite all tests in Mockery.

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

- Refactored files
- Added mockery support
- Rewritten tests with mockery

Fixes #448

### CODE REVIEW

File: cmd/setDelegation.go [hunk 1]
```diff
@@ -10,113 +12,109 @@ import (
...
 		MethodName:      "setDelegationAcceptance",
-		Parameters:      []interface{}{status},
+		Parameters:      []interface{}{delegationInput.Status},
 	}
 
-	if stakerInfo.AcceptDelegation == status {
-		log.Infof("Delegation status already set to %t", status)
-		return nil
+	if stakerInfo.AcceptDelegation == delegationInput.Status {
+		log.Infof("Delegation status already set to %t", delegationInput.Status)
```
Comment (ashish10677) [inline cmd/setDelegation.go:83]: Add tests

File: cmd/setDelegation.go [hunk 2]
```diff
@@ -10,113 +12,109 @@ import (
...
+	razorUtils.WaitForBlockCompletion(client, txn.String())
+}
+
+func (*UtilsStruct) SetDelegation(client *ethclient.Client, config types.Configurations, delegationInput types.SetDelegationInput) (common.Hash, error) {
+
+	stakerInfo, err := razorUtils.GetStaker(client, delegationInput.Address, delegationInput.StakerId)
 	if err != nil {
-		log.Error("Error in fetching staker info")
-		return err
+		return core.NilHash, nil
```
Comment (ashish10677) [inline cmd/setDelegation.go:67]: Add test

File: cmd/stake.go [hunk 3]
```diff
@@ -12,12 +13,9 @@ import (
 	"github.com/spf13/cobra"
 )
 
-var razorUtils utilsInterface
-var tokenManagerUtils tokenManagerInterface
-var transactionUtils transactionInterface
-var stakeManagerUtils stakeManagerInterface
-
-//var utilsStructInterface structUtilsInterface
+//var razorUtils utilsInterface
```
Comment (ashish10677) [inline cmd/stake.go:16]: Remove these lines

File: package-lock.json [hunk 4]
```diff
@@ -5,6 +5,7 @@
   "requires": true,
   "packages": {
     "": {
+      "name": "razor-go",
```
Comment (ashish10677) [inline package-lock.json:8]: Remove this

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: cmd/claimBounty.go [hunk 5]
```diff
@@ -100,33 +93,33 @@ func claimBounty(config types.Configurations, client *ethclient.Client, redeemBo
...
 		}
 	}
 	return core.NilHash, err
 }
 
 func init() {
-	razorUtils = Utils{}
-	transactionUtils = TransactionUtils{}
+	razorUtils = &Utils{}
+	cmdUtils = &UtilsStruct{}
```
[the remark this answers is not in the document]
Comment (Yashk767) [reply cmd/claimBounty.go:119]: Can we have an issue for this and resolve it there?

File: cmd/propose.go [hunk 6]
```diff
@@ -141,18 +139,18 @@ func getBiggestStakeAndId(client *ethclient.Client, address string, epoch uint32
...
+func (*UtilsStruct) GetIteration(client *ethclient.Client, proposer types.ElectedProposer) int {
 	for i := 0; i < 10000000; i++ {
 		proposer.Iteration = i
-		isElected := utilsStruct.proposeUtils.isElectedProposer(client, proposer, utilsStruct)
+		isElected := cmdUtils.IsElectedProposer(client, proposer)
 		if isElected {
 			return i
 		}
 	}
 	return -1
```
[the remark this answers is not in the document]
Comment (Yashk767) [reply cmd/propose.go:150]: Tests are passing on local but here they are failing due to time taken to run this test. Should we still include ?

**Comment by ashish10677** [comment]
@Yashk767 tests are still failing, can you remove the test for getIteration and push the code?

**Review by ashish10677 — APPROVED** [review]
LGTM

**Pull request merged**